samples/Rewrite/Structure2/UrlRewriteMiddleware.cs

```diff
@@ -0,0 +1,136 @@
+        {
+
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            foreach (Rule rule in _options.Rules)
+            {
+                // 1. Check if the certain component of the path matches with the initial match.
+                // TODO logic for flags.
+                // TODO dont initialize the regex here for every request, have it initialized in 
+                //      either the parsing on lazily initialize and save here
+                Regex compare = new Regex(rule.InitialRule.Variable);
+                Match results = compare.Match(context.Request.Path.ToString());
+                // If the match came back negative (the rule xor'd with whether or not it is inverted)
+                // go to the next rule.
+                if (!(results.Success ^ rule.InitialRule.Invert))
+                {
+                    continue;
+                }
+                // 2. Go through all conditions and compare them to the created string
+                Match previous = Match.Empty;
+                var i = 0;
+                for (i = 0; i < rule.Conditions.Count; i++)
+                {
+                    Condition condition = rule.Conditions[i];
+                    string concatTestString = CollectTestStringFromContext(condition.TestStringSegments, context, previous, results);
+                    bool pass = false;
+                    switch (condition.ConditionRegex.Type)
+                    {
+                        case ConditionType.FileTest:
+                            pass = CheckFileCondition(concatTestString, condition, context);
+                            break;
+                        case ConditionType.IntComp:
+                            pass = CheckIntCondition(concatTestString, condition, context);
+                            break;
+                        case ConditionType.StringComp:
+                            pass = CheckStringCondition(concatTestString, condition, context);
+                            break;
+                        case ConditionType.Regex:
+                            previous = Regex.Match(concatTestString, condition.ConditionRegex.Variable);
+                            pass = previous.Success;
+                            break;
+                    }
+                    if (!(pass ^ condition.ConditionRegex.Invert))
+                    {
+                        break;
+                    }
+                }
+                if (i < rule.Conditions.Count)
+                {
+                    continue;
+                }
+                // at this point, our rule passed, we can now apply the on match function
+                string result = CollectTestStringFromContext(rule.OnMatch, context, previous, results);
+                // for now just replace the path, TODO add flag options here
+                context.Request.Path = new PathString(result);
```

Comment:
Isn't there a way to stop once a rule is evaluate? I remember mod_rewrite having the option to either continue or stop once a rule is evaluate successfully